### ISSUE TITLE
HMR doesn't work in projects that use Plugins that use workers

### ISSUE THREAD

**Issue by shiv19** [NativeScript/nativescript-dev-webpack#981]
https://github.com/EddyVerbruggen/nativescript-particle/issues/7
[NativeScript6 Particle Plugin HMR issue.zip](https://github.com/NativeScript/nativescript-dev-webpack/files/3384516/NativeScript6.Particle.Plugin.HMR.issue.zip)

HMR doesn't work when using nativescript-particle plugin.

This problem did not exist in older versions of dev-webpack

**Comment by DimitarTachev** [comment]
Hi @shiv19,

Thanks for reporting that!

In `nativescript@5.4` we had [an issue with the updates of static resources during HMR LiveSync](https://github.com/NativeScript/nativescript-cli/issues/4607). The CLI was always __refrehing__ the application on HMR change and files like images, fonts and databases were not properly updated with HMR sync. In this way, the users were forced to stop the command and re-run it when they need to update such file.

In order to fix that, starting from `nativescript@6.0.0` we are __restarting__ the application when the Webpack process emits a file __different than a hot update__. In this way, the updates of static resources that are included in the app through the CopyWebpackPlugin are causing an app restart and the app is properly updated.

The issue with apps using the Worker plugin is caused by the following behavior:
1) The Worker plugin is always emitting [an additional file](https://github.com/NativeScript/worker-loader/blob/master/src/NativeScriptWorkerPlugin.js#L24) (`__worker-chunks.json`) in the Webpack compilation. This is not a hot update file and the CLI is restarting the app in order to apply it.
2) The Worker loader is [creating a child Webpack](https://github.com/NativeScript/worker-loader/blob/master/src/index.js#L85) process for each worker file and these Webpack processes are emitting `[hash].worker.js` files. These files are also triggering an app restart as they are not hot updates.

In other words, the Worker plugin and loader should stop emitting these files on each change and we will move this issue in the [Worker loader repository](https://github.com/NativeScript/worker-loader). Unfortunately, the above-mentioned point (2) does not have an easy fix and most probably it will be a known limitation for `nativescript@6.0.0` till we release a better version of the Worker plugin.
